lib/oscap_tests.pm

```diff
@@ -123,186 +711,541 @@ sub upload_logs_reports {
-
     # Check the oscap version information for reference
     $out = script_output("oscap -V");
     record_info("oscap version", "\"# oscap -V\" returns:\n $out");
+
+    # Get the tests configuration file from repository and set global configuration varables
+    get_tests_config();
+    push(@test_run_report, "[configuration]");
+    $out = script_output("date", quiet => 1);
+    push(@test_run_report, "date = $out");
+    push(@test_run_report, "profile_ID = $profile_ID");
+    push(@test_run_report, "ansible_profile_file_name = $ansible_profile_ID");
+    push(@test_run_report, "use_content_type = $use_content_type");
+    push(@test_run_report, "remove_rules_missing_fixes = $remove_rules_missing_fixes");
+    push(@test_run_report, "use_exclusions = $use_exclusions");
+    push(@test_run_report, "evaluate_count = $evaluate_count");
+
+    # Replace original ds and xccdf files whith downloaded from local repository
+    set_ds_file_name();
+    push(@test_run_report, "sle_version = $sle_version");
+    push(@test_run_report, "ssg_sle_ds = $ssg_sle_ds");
+    push(@test_run_report, "ssg_sle_xccdf = $ssg_sle_xccdf");
+    my $arch = get_var 'ARCH';
+    push(@test_run_report, "os_arch = $arch");
+    my $type;
+    if ($ansible_remediation == 1) {
+        $type = 'ansible';
+    }
+    else {
+        $type = 'bash';
+    }
+    push(@test_run_report, "remediation_type = $type");
+    my $build_url = get_var 'CASEDIR';
+    push(@test_run_report, "build_url = $build_url");
+    my $iso_name = get_var 'ISO';
+    push(@test_run_report, "iso_name = $iso_name");
+    my $full_name = get_var 'NAME';
+    push(@test_run_report, "full_name = $full_name");
+    my $schedule = get_var 'YAML_SCHEDULE';
+    push(@test_run_report, "schedule = $schedule");
+
+    unless (is_opensuse) {
+        # Some Packages require PackageHub repo is available
+        return unless is_phub_ready();
+        add_suseconnect_product(get_addon_fullname('phub'));
+        # Need to use pyython3.1x
+        add_suseconnect_product(get_addon_fullname('python3'));
+        # On SLES 12 ansible packages require depencies located in sle-module-public-cloud
+        add_suseconnect_product(get_addon_fullname('pcm'), (is_sle('<15') ? '12' : undef)) if is_sle('<15');
+    }
+
+    # If required ansible remediation
+    if ($ansible_remediation == 1) {
+        my $pkgs = 'ansible';
+        # Need to update SLES to fix issues with STIG playbook
+        record_info("Update", "Updaiting SLES");
+        zypper_call("up", timeout => 1800);
+        zypper_call "in $pkgs sudo";
+        # Record the pkgs' version for reference
+        my $out = script_output("zypper se -s $pkgs", quiet => 1);
+        record_info("$pkgs Pkg_ver", "$pkgs packages' version:\n $out");
+        #install ansible.posix
+        assert_script_run("ansible-galaxy collection install ansible.posix");
+    }
+    if ($remove_rules_missing_fixes == 1 or $use_content_type == 3) {
```

Comment:
I'm not that expert in Perl, but we always (or most of the time in the repo) use `||` instead of `or` I think that is because of the precedence, that avoid some tricky mistakes.

Reply:
Can use both, but "or" has low precedence: 
!, &&, ||, and ^ have high precedence such that they are useful in constructing an expression; not, and, or, and xor have low precedence such that they are useful for flow control between what are essentially different expressions.

Fixed it to:
(($use_content_type == 2) or ($use_content_type == 3))